cli/lib/kontena/cli/stacks/common.rb

```diff
@@ -110,6 +110,11 @@ def require_config_file(filename)
     def generate_volumes(yaml_volumes)
       return [] unless yaml_volumes
       yaml_volumes.map do |name, config|
+        if config['external'].is_a?(TrueClass)
+          config['external'] = name
+        elsif config['external']['name']
```

Comment:
Might also make sense to support `external: false`... but it should be mapped to `{"external": null }` (or better yet, omit it entirely) in the API.